platform/core/src/runs/TransactingMutationListener.js

```diff
@@ -38,9 +44,28 @@ define([], function () {
                     transactionService.startTransaction();
                 }
 
+                if (!MUTATION_TRACKER.has(domainObject)) {
+                    MUTATION_TRACKER.set(domainObject, domainObject
+                        .getCapability('mutation')
+                        .listen(function () {})
```

Comment:
The synchronizing mutation listener keeps a domain object in sync as long as something is listening to mutation on that specific domain object. This is a fill that ensures if you mutate an object it will stay in sync from that point until you close the transaction.  Depending on which instance of a domain object persist actually gets called on (that's one part of the transaction flow I couldn't isolate), this could theoretically be different.

Now that's just my rationale, this might not be a necessary part of the solution, need to nail it down.